From 6a49928db006075c09c18e363ff49ada5c053a71 Mon Sep 17 00:00:00 2001
From: Dominik Behr <dbehr@chromium.org>
Date: Thu, 31 Oct 2019 13:26:31 -0700
Subject: [PATCH] CHROMIUM: do not initialize/destroy locale for strtod_l

It is not used anyway, and not refcounted so it causes crashes
with multiple Vulkan instances.

BUG=b:139651193
TEST=am instrument -w org.skia.skqp does not crash

Change-Id: I64bef666ab2971c5c2ae4f9b0bb921ba25005125
Signed-off-by: Dominik Behr <dbehr@chromium.org>
---
 src/amd/vulkan/radv_device.c  | 4 ----
 src/intel/vulkan/anv_device.c | 4 ----
 2 files changed, 8 deletions(-)

diff --git a/src/amd/vulkan/radv_device.c b/src/amd/vulkan/radv_device.c
index 68c6b647718..2e1cc10c422 100644
--- a/src/amd/vulkan/radv_device.c
+++ b/src/amd/vulkan/radv_device.c
@@ -584,8 +584,6 @@ VkResult radv_CreateInstance(
 		return vk_error(instance, result);
 	}
 
-	_mesa_locale_init();
-
 	VG(VALGRIND_CREATE_MEMPOOL(instance, 0, false));
 
 	radv_handle_per_app_options(instance, pCreateInfo->pApplicationInfo);
@@ -610,8 +608,6 @@ void radv_DestroyInstance(
 
 	VG(VALGRIND_DESTROY_MEMPOOL(instance));
 
-	_mesa_locale_fini();
-
 	vk_debug_report_instance_destroy(&instance->debug_report_callbacks);
 
 	vk_free(&instance->alloc, instance);
diff --git a/src/intel/vulkan/anv_device.c b/src/intel/vulkan/anv_device.c
index f44b046cf5d..bc4204e27a0 100644
--- a/src/intel/vulkan/anv_device.c
+++ b/src/intel/vulkan/anv_device.c
@@ -672,8 +672,6 @@ VkResult anv_CreateInstance(
    instance->pipeline_cache_enabled =
       env_var_as_boolean("ANV_ENABLE_PIPELINE_CACHE", true);
 
-   _mesa_locale_init();
-
    VG(VALGRIND_CREATE_MEMPOOL(instance, 0, false));
 
    *pInstance = anv_instance_to_handle(instance);
@@ -703,8 +701,6 @@ void anv_DestroyInstance(
 
    vk_debug_report_instance_destroy(&instance->debug_report_callbacks);
 
-   _mesa_locale_fini();
-
    vk_free(&instance->alloc, instance);
 }
 
-- 
2.24.0.rc0.303.g954a862665-goog

